src/main/java/ActionManager.java

```diff
@@ -0,0 +1,163 @@
+            tasks[taskCount] = new Event(eventName, eventFrom, eventTo);
+            System.out.println("Deadline task added!\n");
+            tasks[taskCount++].printTask();
+        } catch (SuBotException e) {
+            String msg = e.getMessage() + "\n" + UsageString.EVENT_USAGE;
+            throw new SuBotException(msg);
+        }
+    }
+
+    /**
+     * Adds a <code>Todo</code> to list of tasks
+     * @param description Task description
+     * @throws SuBotException If description is blank
+     */
+    private static void addTodo(String description) throws SuBotException{
+        try {
+            Parser.validateToDo(description); //Doesn't need a parser
+            tasks[taskCount] = new ToDo(description);
+            System.out.println("Todo task added!\n");
+            tasks[taskCount++].printTask();
+        } catch (SuBotException e) {
+            String msg = e.getMessage() + "\n" + UsageString.TODO_USAGE;
+            throw new SuBotException(msg);
+        }
+    }
+
+    /**
+     * Adds a <code>Deadline</code> to list of tasks.
+     * Note that <code>&lt;time&gt;</code> is treated as a <code>String</code>
+     * @param argString Format: <code> &lt;name&gt; /by &lt;time&gt;</code>
+     * @throws SuBotException If invalid format or any empty argument
+     */
+    private static void addDeadline(String argString) throws SuBotException {
+        try {
+            String[] argv = Parser.parseDeadline(argString);
+            String dlName  = argv[0];
+            String dlTime  = argv[1];
+            tasks[taskCount] = new Deadline(dlName, dlTime);
+            System.out.println("Deadline task added!\n");
+            tasks[taskCount++].printTask();
+        } catch (SuBotException e) {
+            String msg = e.getMessage() + "\n" + UsageString.DEADLINE_USAGE;
+            throw new SuBotException(msg);
+        }
+    }
+
+    /**
+     * Processes user's input command.
+     * This method takes the first word in <code>command</code>
+     * to determine the action needed.
+     * The remaining (if available) will be passed to seperated methods
+     * for further processing.
+     * @param command Input command
+     */
+      static void process(String command) {
+        //Split `command` into `action` and `argString`
+        String[] argv = command.split(" ", 2);
+        //argv will have at least length of 1 because that's how split() works
+        String action = argv[0];
+
+        String argString;
+        int argc = argv.length;
+        if (argc == 2) {
+            argString = argv[1]; //For commands with 1 or more options
+        } else {
+            argString = ""; //For 0-option commands (E.g. list)
+        }
+        try {
+            switch (action) {
+            case "bye":
+                SuBOT.isRunning = false; //Halt signal for parent class
+                break;
+            case "list":
+                listAllTask();
+                break;
+            case "mark":
```

Comment:
Is there meant to be a "break" here? Otherwise, it would be good to indicate that this was intended using "//Fallthrough"